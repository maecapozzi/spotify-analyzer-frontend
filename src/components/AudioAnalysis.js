import React from 'react'
import injectSheet from 'react-jss'
import PropTypes from 'prop-types'

const styles = {
  value: {
    color: '#00C457'
  }
}

const AudioAnalysis = ({classes, type, value}) => (
  <div>
    <p>
      <span>{type}: </span>
      <span className={classes.value}>{value}</span>
    </p>
  </div>
)

AudioAnalysis.propTypes = { 
  type: PropTypes.string,
  value: PropTypes.number
}

export default injectSheet(styles)(AudioAnalysis)
