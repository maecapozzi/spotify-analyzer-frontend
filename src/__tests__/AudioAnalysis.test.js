import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

import AudioAnalysis from '../components/AudioAnalysis'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const audioAnalysisProps = {
    type: 'Time Signature',
    value: '4/4'
  }

  const component = shallow(<AudioAnalysis {...audioAnalysisProps} />)
  expect(component).toMatchSnapshot()
})
