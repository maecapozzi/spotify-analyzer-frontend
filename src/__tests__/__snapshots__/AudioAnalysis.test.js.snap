// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(AudioAnalysis)
    type="Time Signature"
    value="4/4"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "value": "AudioAnalysis-value-0-1",
      },
      "type": "Time Signature",
      "value": "4/4",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "value": "AudioAnalysis-value-0-1",
        },
        "type": "Time Signature",
        "value": "4/4",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
